cpp/src/neighbors/detail/cagra/graph_core.cuh

```diff
@@ -488,6 +490,53 @@ void shift_array(T* array, uint64_t num)
     array[i] = array[i - 1];
   }
 }
+
+// RAII wrapper for allocating memory with Transparent HugePage
+struct mmap_owner {
```

Comment:
We have already one version here https://github.com/rapidsai/cuvs/blob/2ca4d3a9ff1972199a0463bf963cd66ef398cdfe/cpp/src/neighbors/detail/cagra/cagra_build.cuh#L443

To avoid duplication, could you move this to a separate header and include that here as well as in `cagra_build.cuh`?